File: app/src/main/java/com/jidesh/erosnowjidesh/ui/main/Adapters/SectionsPagerAdapter.java
package com.jidesh.erosnowjidesh.ui.main.Adapters;

import android.content.Context;

import androidx.annotation.Nullable;
import androidx.annotation.StringRes;
import androidx.fragment.app.Fragment;
import androidx.fragment.app.FragmentManager;
import androidx.fragment.app.FragmentPagerAdapter;

import com.jidesh.erosnowjidesh.R;
import com.jidesh.erosnowjidesh.ui.main.Fragments.Favourite_fragment;
import com.jidesh.erosnowjidesh.ui.main.Fragments.PlaceholderFragment;

/**
 * A [FragmentPagerAdapter] that returns a fragment corresponding to
 * one of the sections/tabs/pages.
 */
public class SectionsPagerAdapter extends FragmentPagerAdapter {

    @StringRes
    private static final int[] TAB_TITLES = new int[]{R.string.tab_text_1, R.string.tab_text_2};
    private final Context mContext;

    public SectionsPagerAdapter(Context context, FragmentManager fm) {
        super(fm);
        mContext = context;
    }

    @Override
    public Fragment getItem(int position) {
        Fragment a=new PlaceholderFragment();
        switch (position) {
            case 1: a= new PlaceholderFragment();
            case 2:a= new Favourite_fragment();
        }
        // getItem is called to instantiate the fragment for the given page.
        // Return a PlaceholderFragment (defined as a static inner class below).
        return a;
    }

    @Nullable
    @Override
    public CharSequence getPageTitle(int position) {
        return mContext.getResources().getString(TAB_TITLES[position]);
    }

    @Override
    public int getCount() {
        // Show 2 total pages.
        return 2;
    }
}

File: app/src/main/java/com/jidesh/erosnowjidesh/ui/main/Adapters/SectionsPagerAdapter.kt
package com.jidesh.erosnowjidesh.ui.main.Adapters

import android.content.Context
import androidx.annotation.StringRes
import androidx.fragment.app.Fragment
import androidx.fragment.app.FragmentManager
import androidx.fragment.app.FragmentPagerAdapter
import com.jidesh.erosnowjidesh.R
import com.jidesh.erosnowjidesh.ui.main.Fragments.Favourite_fragment
import com.jidesh.erosnowjidesh.ui.main.Fragments.PlaceholderFragment

class SectionsPagerAdapter(private val mContext: Context, fm: FragmentManager) : FragmentPagerAdapter(fm) {
    override fun getItem(position: Int): Fragment {
       var a: Fragment = PlaceholderFragment()
        when (position) {
            0 -> {

              a  = PlaceholderFragment()

            }
            1 -> {
               a = Favourite_fragment()
            }
        }

        return a
    }

    override fun getPageTitle(position: Int): CharSequence? {
        return mContext.resources.getString(TAB_TITLES[position])
    }

    override fun getCount(): Int { // Show 2 total pages.
        return 2
    }

    companion object {
        @StringRes
        private val TAB_TITLES = intArrayOf(R.string.tab_text_1, R.string.tab_text_2)
    }

}